src/analyzer/analyzersilence.cpp

```diff
@@ -43,34 +41,60 @@ bool AnalyzerSilence::initialize(TrackPointer pTrack,
     }
 
     m_iFramesProcessed = 0;
-    m_bPrevSilence = true;
     m_iSignalStart = -1;
     m_iSignalEnd = -1;
 
     return true;
 }
 
-bool AnalyzerSilence::processSamples(const CSAMPLE* pIn, const int iLen) {
-    for (int i = 0; i < iLen; i += mixxx::kAnalysisChannels) {
-        // Compute max of channels in this sample frame
-        CSAMPLE fMax = CSAMPLE_ZERO;
-        for (SINT ch = 0; ch < mixxx::kAnalysisChannels; ++ch) {
-            CSAMPLE fAbs = fabs(pIn[i + ch]);
-            fMax = math_max(fMax, fAbs);
+// static
+SINT AnalyzerSilence::findFirstSoundInChunk(const CSAMPLE* pIn, SINT iLen) {
+    SINT i = 0;
+    for (; i < iLen; ++i) {
+        if (fabs(pIn[i]) >= kSilenceThreshold) {
+            break;
         }
+    }
+    return i;
+}
 
-        bool bSilence = fMax < m_fThreshold;
-
-        if (m_bPrevSilence && !bSilence) {
-            if (m_iSignalStart < 0) {
-                m_iSignalStart = m_iFramesProcessed + i / mixxx::kAnalysisChannels;
-            }
-        } else if (!m_bPrevSilence && bSilence) {
-            m_iSignalEnd = m_iFramesProcessed + i / mixxx::kAnalysisChannels;
+// static
+SINT AnalyzerSilence::findLastSoundInChunk(const CSAMPLE* pIn, SINT iLen) {
+    SINT i = iLen - 1;
+    for (; i >= 0; --i) {
+        if (fabs(pIn[i]) >= kSilenceThreshold) {
+            break;
         }
+    }
+    return i;
+}
```

Comment:
Even this is full of boilerplate and could be deduplicated even more.
```suggestion
/// returns a std::reverse_iterator
auto AnalyzerSilence::findLastSoundInChunk(std::span<const CSAMPLE> samples) {
    return first_sound(samples.rbegin()), samples.rend());
}
```